core/gui/src/app/dashboard/component/user/list-item/list-item.component.ts

```diff
@@ -142,27 +141,18 @@ export class ListItemComponent implements OnInit, OnChanges {
     }
   }
 
-  public onClickDownload(): void {
+  public onClickDownload = (): void => {
+    if (!this.entry.id) return;
+
     if (this.entry.type === "workflow") {
-      if (this.entry.id) {
-        this.workflowPersistService
-          .retrieveWorkflow(this.entry.id)
-          .pipe(untilDestroyed(this))
-          .subscribe(data => {
-            const workflowCopy: Workflow = {
-              ...data,
-              wid: undefined,
-              creationTime: undefined,
-              lastModifiedTime: undefined,
-              readonly: false,
-            };
-            const workflowJson = JSON.stringify(workflowCopy.content);
-            const fileName = workflowCopy.name + ".json";
-            this.fileSaverService.saveAs(new Blob([workflowJson], { type: "text/plain;charset=utf-8" }), fileName);
-          });
-      }
+      this.downloadService
+        .downloadWorkflow(this.entry.id, this.entry.workflow.workflow.name)
+        .pipe(untilDestroyed(this))
+        .subscribe();
+    } else if (this.entry.type === "dataset") {
+      this.downloadService.downloadDataset(this.entry.id, this.entry.name).pipe(untilDestroyed(this)).subscribe();
```

Comment:
add some notifications once download successfully/failed?

Reply:
The notifications are already implemented inside the DownloadService. I haven't changed any logic for notifications. If you want to add more notifications, it should be in a different PR.